File: aview/app/src/main/java/com/example/aview/MainActivity.java
package com.example.aview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    String[] general = {"이순신", "강감찬", "김유신", "을지문덕", "권율"};
    ArrayList<String> color;
    ListView lv;
    ArrayAdapter<String> adapter;
    AdapterView.OnItemClickListener icl;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        color = new ArrayList<String>();
        color.add("빨강색");
        color.add("파란색");
        color.add("노랑색");
        color.add("하늘색");
        color.remove(2);

        lv = (ListView) findViewById(R.id.list);
        //adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, general);
        adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, color);

        lv.setAdapter(adapter);

        icl = new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Toast.makeText(getApplicationContext(), "번호= " + i + " 내용= " + color.get(i), Toast.LENGTH_LONG).show();
            }
        };
        lv.setOnItemClickListener(icl);
    }
}

File: SecretDiary/app/src/main/java/fastcampus/aop/part2/secretdiary/MainActivity.kt
package fastcampus.aop.part2.secretdiary

import android.content.Context
import android.content.Intent
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.NumberPicker
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.widget.AppCompatButton
import androidx.core.content.edit

class MainActivity : AppCompatActivity() {

    private val numberPicker1 : NumberPicker by lazy{
        findViewById<NumberPicker>(R.id.numberPicker1)
            .apply {
                minValue = 0
                maxValue = 9
            }
    }

    private val numberPicker2 : NumberPicker by lazy{
        findViewById<NumberPicker>(R.id.numberPicker2)
            .apply {
                minValue = 0
                maxValue = 9
            }
    }

    private val numberPicker3 : NumberPicker by lazy{
        findViewById<NumberPicker>(R.id.numberPicker3)
            .apply {
                minValue = 0
                maxValue = 9
            }
    }

    private val openButton: AppCompatButton by lazy {
        findViewById<AppCompatButton>(R.id.openButton)
    }

    private val changePasswordButton: AppCompatButton by lazy {
        findViewById<AppCompatButton>(R.id.changePasswordButton)
    }

    private var changePasswordMode = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //lazy를 했기에 초기화를 위해 사용
        numberPicker1
        numberPicker2
        numberPicker3

        openButton.setOnClickListener {

            if (changePasswordMode) {
                Toast.makeText(this, "비밀번호 변경중", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            //프리퍼런스라는 파일을 다른 앱과 공유하는 메소드이지만 MODE_PRIVATE를 이용하여 자신만 사용
            val passwordPreferences = getSharedPreferences("password", Context.MODE_PRIVATE)

            val passwordFromUser = "${numberPicker1.value}${numberPicker2.value}${numberPicker3.value}"

            if (passwordPreferences.getString("password", "000").equals(passwordFromUser)){
                //패스워드가 일치(잠금해제)
                startActivity(Intent(this, DiaryActivity::class.java))
            } else {
                //패스워드 불일치
                showErrorAlertDialog()
            }
        }

        changePasswordButton.setOnClickListener {

            val passwordPreferences = getSharedPreferences("password", Context.MODE_PRIVATE)
            val passwordFromUser = "${numberPicker1.value}${numberPicker2.value}${numberPicker3.value}"

            if (changePasswordMode) {
                //번호를 저장
                passwordPreferences.edit(true) {
                    putString("password", passwordFromUser)
                }

                changePasswordMode = false
                changePasswordButton.setBackgroundColor(Color.BLACK)

            } else {
                //changePasswordMode 활성화 :: 비밀번호가 맞는지를 체크
                if (passwordPreferences.getString("password", "000").equals(passwordFromUser)){

                    changePasswordMode = true
                    Toast.makeText(this, "변경할 비밀번호 입력", Toast.LENGTH_SHORT).show()

                    changePasswordButton.setBackgroundColor(Color.RED)

                } else {
                    //패스워드 불일치
                    showErrorAlertDialog()
                }
            }

        }

    }

    private fun showErrorAlertDialog(){
        AlertDialog.Builder(this).setTitle("불일치").setMessage("비밀번호가 잘못되었습니다").setPositiveButton("확인") { _, _ -> }
            .show()
    }
}